### PR TITLE
Remove deprecated billingID from baseIPN

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Remove and replace with supported function.

Before
----------------------------------------
Deprecated function in use.

After
----------------------------------------
Deprecated function gone.

Technical Details
----------------------------------------
Step towards deprecating baseIPN.

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@mattwire what do you think about moving the noise to within CRM_Core_BAO_LocationType::getBilling()

I think having a billing ID is more-or-less compulsory & I think it would be OK to be noisy in that frequently-called function & have less handling elsewhere

File: CRM/Core/Payment/BaseIPN.php [hunk 1]
```diff
@@ -483,23 +483,6 @@ public function completeTransaction(&$input, &$ids, &$objects, $transaction = NU
...
   }
 
-  /**
-   * Get site billing ID.
-   *
-   * @param array $ids
-   *
-   * @return bool
-   */
-  public function getBillingID(&$ids) {
```
Comment (eileenmcnaughton) [inline CRM/Core/Payment/BaseIPN.php:493]: I think we should deprecate rather than remove this

**Comment by mattwire** [comment]
@eileenmcnaughton I don't really see the point in those warnings about having a billing location. Most payment processor extensions don't do that as the expectation that there is a billing location seems pretty fundamental. There were a lot of instances where things were hardcoded to "5" in the past but the assumption that there may not be a billing address does not seem valid.
Let me know what you think please?

File: CRM/Core/Payment/BaseIPN.php (hunk 1 above)
Comment (mattwire) [reply CRM/Core/Payment/BaseIPN.php:493]: @eileenmcnaughton That's fine. Please let me know about the other comment I made and I will update this

**Comment by eileenmcnaughton** [comment]
@mattwire the debug seems extraneous to me - I'm happy to see it gone

**Comment by eileenmcnaughton** [comment]
ie I think it's OK to assume ```CRM_Core_BAO_LocationType::getBilling()``` always returns what wee want

**Comment by mattwire** [comment]
Ok I think this should be ok now

File: CRM/Core/Payment/AuthorizeNetIPN.php [hunk 2]
```diff
@@ -230,7 +230,9 @@ public function getInput(&$input, &$ids) {
       $input['trxn_id'] = md5(uniqid(rand(), TRUE));
     }
 
-    if (!$this->getBillingID($ids)) {
+    $ids['billing'] = CRM_Core_BAO_LocationType::getBilling();
```
Comment (eileenmcnaughton) [inline CRM/Core/Payment/AuthorizeNetIPN.php:233]: @mattwire so what I thought we agreed was to remove 
```
 if (!$ids['billing']) {
      \Civi::log()->error(ts('Please set a location type of %1', [1 => 'Billing']));
      return FALSE;
    }
```

entirely

Comment (mattwire) [reply CRM/Core/Payment/AuthorizeNetIPN.php]: @eileenmcnaughton Ok, see now?

**Comment by eileenmcnaughton** [comment]
@mattwire style warnings

**Pull request merged**